descqa/apparent_mag_func_test.py

```diff
@@ -133,38 +149,47 @@ def run_on_single_catalog(self, catalog_instance, catalog_name, output_dir):
         if not mag_field_key:
             return TestResult(skipped=True, summary='Catalog is missing requested quantity: {}'.format(self.possible_mag_fields))
 
+        # check to see if catalog is a light cone
+        # this is required since we must be able to calculate the angular area
+        try:
+            if not catalog_instance.lightcone:
+                return TestResult(skipped=True, summary="Catalog is not a light cone.")
+        except AttributeError:
+            if not catalog_instance.has_quantity('ra') or not catalog_instance.has_quantity('dec'):
+                return TestResult(skipped=True, summary="'ra' and/or 'dec' not available to compute sky area")
+
+        # check to see the angular area if an attribute of the catalog
+        try:
+            sky_area = catalog_instance.sky_area
+        except AttributeError:
+            sky_area = get_sky_area(catalog_instance) # compute area from ra and dec
```

Comment:
Reorganize the logic a little bit for lines 152-165. In particular, checking ra/dec existence is more relevant for sky area, not for light cone. 

```python
	    # check to see if catalog is a light cone
        # this is required since we must be able to calculate the angular area
        # if attribute `lightcone` does not exist, allow the catalog to proceed
        if not getattr(catalog_instance, 'lightcone', True):
            return TestResult(skipped=True, summary="Catalog is not a light cone.")

        # obtain or calculate sky area
        sky_area = getattr(catalog_instance, 'sky_area', None)
		if sky_area is None:
            if not catalog_instance.has_quantities(['ra', 'dec']):
                return TestResult(skipped=True, summary="'ra' and/or 'dec' not available to compute sky area")
		    sky_area = get_sky_area(catalog_instance) # compute area from ra and dec
```

Reply:
@yymao  Ah, thanks, using getattr is much better. Will make a note of it. Change is made and tested.